compiler/src/iree/compiler/Preprocessing/Common/PadToIntrinsics.cpp

```diff
@@ -198,15 +504,26 @@ struct PadToIntrinsicsPass final
           return GPUMatmulShapeType{mSize, nSize, kSize, aType, bType, cType};
         });
 
-    SmallVector<linalg::Conv2DNhwcHwcfOp> targetOps;
-    rootOp->walk(
-        [&](linalg::Conv2DNhwcHwcfOp convOp) { targetOps.push_back(convOp); });
+    SmallVector<linalg::LinalgOp> targetOps;
+    rootOp->walk([&](linalg::LinalgOp linalgOp) {
```

Comment:
note that this is unsafe - root walks are not allowed outside of codegen as this will walk into any executables that already exist and mess with them - instead you should loop over FunctionOpInterface ops and walk those

(if a user has partially compiled the program, used custom ops/dispatches, tk, etc there will be executables in the input)

Reply:
Hey Ben, I modified the code to be FunctionOpInterface so it will just walk on funcOpInterface operations. I also added the more "local" look up and exit if the number of targetAttrs found is none-1.